tests/unit/test_charm.py

```diff
@@ -25,9 +25,20 @@ def setup_postgres_relation(harness):
     )
 
```

Comment:
A note on mocking, we don't want to mock the method `mocked_sql_migration`. What we really want to mock is the 'container.exec` command in it. The reason is that if we end up refactoring the charm code one day and change the place where the sql migrations are run, now all the unit tests fail even though the behaviour may have remained the same. If the code is refactored but the behaviour remained the same, unit tests should still pass.

Reply:
The same is true of `mocked_update_container`

Reply:
We should aim at removing those mocks. This may end up in our tests being too complex. This is good because it'll reflect the fact that the charms are also too complex and puts pressure for a better design